### PR TITLE
SMS77 Notification Channel

### PR DESCRIPTION
**Pull request by mxschll**
Notification channel for [SMS77.io](https://www.sms77.io/)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+language: php
```
Comment (atymic) [inline .travis.yml:1]: Can prob drop this if using GH actions :)

File: README.md [hunk 2]
```diff
@@ -1,17 +1,114 @@
-# New Notification Channels
+# SMS77 notifications channel for Laravel
 
-### Suggesting a new channel
-Have a suggestion or working on a new channel? Please create a new issue for that service.
+[![Latest Version on Packagist](https://img.shields.io/packagist/v/laravel-notification-channels/:package_name.svg?style=flat-square)](https://packagist.org/packages/laravel-notification-channels/:package_name)
```
Comment (atymic) [inline README.md:3]: find/replace all of these badges please

File: README.md [hunk 3]
```diff
@@ -1,17 +1,114 @@
...
+
+## Installation
+
+This package can be installed via composer:
+
+```composer require laravel-notification-channels/sms77```
+
+### Setting up the SMS77 service
+
+1. Create an account and get the API key [here](https://www.sms77.io/de/)
```
Comment (atymic) [inline README.md:36]: maybe remove the `/de` from the url so it uses the right lang for the user :)

File: README.md [hunk 4]
```diff
@@ -1,17 +1,114 @@
...
+
+2. Add the API key to the `services.php` config file:
+
+	```php
+	// config/services.php
+	...
+	'sms77' => [
+		'api_key' => env('SMS77_API_KEY', 'YOUR API KEY HERE')
```
Comment (atymic) [inline README.md:44]: I wouldn't suggest putting the default in the config file here
```suggestion
		'api_key' => env('SMS77_API_KEY')
```

File: phpunit.xml.dist [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+            <directory>tests</directory>
+        </testsuite>
+    </testsuites>
+    <filter>
+        <whitelist>
+            <directory suffix=".php">src/</directory>
+        </whitelist>
+    </filter>
+    <logging>
+        <log type="tap" target="build/report.tap"/>
```
Comment (atymic) [inline phpunit.xml.dist:23]: remove this line (invalid config in new phpunit)

File: src/SMS77.php [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+namespace NotificationChannels\SMS77;
+
+use Exception;
+use GuzzleHttp\Client as HttpClient;
+use GuzzleHTtp\Exception\ClientException;
+use NotificationChannels\SMS77\Exceptions\CouldNotSendNotification;
+
+/**
+ * Class SMS77.
```
Comment (atymic) [inline src/SMS77.php:11]: can remove this docblock

File: src/SMS77.php [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    public function sendRequest(string $endpoint, array $params)
+    {
+        if (empty($this->api_key)) {
+            throw CouldNotSendNotification::apiKeyNotProvided();
+        }
+
+        $request_url = $this->api_url.$endpoint;
+
+        try {
+            return $this->httpClient()->post($request_url, [
```
Comment (atymic) [inline src/SMS77.php:112]:
```suggestion
            return $this->httpClient()->post($this->api_url.$endpoint, [
```

Save setting the var above :)

File: src/SMS77.php [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        return $this->sendRequest('sms', $params);
+    }
+
+    public function sendRequest(string $endpoint, array $params)
+    {
+        if (empty($this->api_key)) {
+            throw CouldNotSendNotification::apiKeyNotProvided();
+        }
+
+        $request_url = $this->api_url.$endpoint;
```
Comment (atymic) [inline src/SMS77.php:109]: Camelcase for var names please

File: src/SMS77Message.php [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+    {
+        return new self($message);
+    }
+
+    /**
+     * Returns if recipient number is given or not.
+     *
+     * @return bool
+     */
+    public function toIsset(): bool
```
Comment (atymic) [inline src/SMS77Message.php:48]: think something a bit more readable like `hasToNumber` would make sense

**Review by atymic — CHANGES_REQUESTED** [review]
Looks good!
Couple of minor changes, then will set a up a repo for the new channel
Thanks!

**Comment by mxschll** [comment]
Thanks for taking your time @atymic and reviewing this :)

File: .travis.yml (hunk 1 above)
Comment (mxschll) [reply .travis.yml:1]: I think Travis needs this line, because when removed, it runs tests for Ruby.

Comment (atymic) [reply .travis.yml:1]: I will disable it on the new repo :)

File: README.md (hunk 2 above)
Comment (atymic) [reply README.md:3]: still missing package names in here

File: src/SMS77.php (hunk 6 above)
Comment (atymic) [reply src/SMS77.php:11]: was not removed

**Review by atymic — APPROVED** [review]
Just a few minor bits. Will create a repo now and add you.

**Comment by atymic** [comment]
Closing, channel approved!

**Pull request closed without merging**